src/main/java/com/salesforce/storm/spout/dynamic/Tools.java

```diff
@@ -49,14 +47,8 @@
      * @return sshallow cloned map that is immutable.
      */
     public static <K,V> Map<K,V> immutableCopy(Map<K,V> sourceMap) {
-        // If we're already dealing with an UnmodifiableMap
-        if (sourceMap instanceof UnmodifiableMap) {
```

Comment:
Interestingly this is NOT the `UnmodifiableMap` used by `Collections.unmodifiableMap()` so this never actually did anything.

Reply:
should we add the correct instance check here?

Reply:
I looked into this, the problem is the actual class is private to the Collections class, so we can’t.

Reply:
For reference http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/util/Collections.java#l1499